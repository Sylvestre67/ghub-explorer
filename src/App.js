import React, { Component } from 'react';

import { withStyles } from '@material-ui/core';

import Grid from '@material-ui/core/Grid';
import Typography from '@material-ui/core/Typography';

import Header from './layout/Header';
import Footer from './layout/Footer';

const styles = theme => ({
	root: {
		height: '100%',
		display: 'flex',
		flexFlow: 'column',
	},
	body: {
		flexGrow: '1',
		backgroundColor: 'white',
	},
	jumbo: {
		textAlign: 'center',
		padding: `${theme.spacing.unit * 3}px 0`,
	},
});

const site_title = `${
	process.env.REACT_APP_WEBSITE_NAME
} Github Repository Search`;

class App extends Component {
	render() {
		const { classes } = this.props;

		return (
			<div className={classes.root}>
				<Header />
				<div className={classes.body}>
					<Grid container>
						<Grid item xs className={classes.jumbo}>
							<Typography variant="h5">{site_title}</Typography>
						</Grid>
					</Grid>

					<Grid container>
						<Grid item xs>
							<Typography>Form</Typography>
						</Grid>
					</Grid>

					<Grid container>
						<Grid item xs>
							<Typography>Results</Typography>
						</Grid>
					</Grid>
				</div>
				<Footer />
			</div>
		);
	}
}

export default withStyles(styles)(App);
