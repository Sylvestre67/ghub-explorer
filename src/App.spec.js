import React from 'react';
import { shallow } from 'enzyme';

import App from './App';
let component;

const props = {};

describe('App', () => {
	beforeAll(() => {
		component = shallow(<App {...props} />);
	});

	it('should render correctly', () => {
		expect(component).toMatchSnapshot();
	});
});
