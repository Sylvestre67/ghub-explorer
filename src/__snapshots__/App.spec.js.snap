// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly 1`] = `
<App
  classes={
    Object {
      "body": "App-body-2",
      "jumbo": "App-jumbo-3",
      "root": "App-root-1",
    }
  }
/>
`;
